kivy/core/image/__init__.py

```diff
@@ -626,28 +658,68 @@ def anim_reset(self, allow_anim):
             self._anim_ev.cancel()
             self._anim_ev = None
 
-        if allow_anim and self._anim_available and self._anim_delay >= 0:
-            self._anim_ev = Clock.schedule_interval(self._anim,
-                                                    self.anim_delay)
-            self._anim()
+        if allow_anim and self._anim_available:
+            if self._durations:
+                if not len(self._durations) == len(self.image.durations):
+                    raise Exception(
+                        'Image frames count not equal durations count')
+                # there are minimum 2 elements otherwise this is constant case
+                # and durations is None, this index is for the next frame
+                self._anim_index = 1
+                self._anim_ev = Clock.schedule_once(self._anim,
+                                                    self._durations[0])
+            elif self._anim_delay >= 0:
+                self._anim_ev = Clock.schedule_interval(self._anim,
+                                                        self._anim_delay)
 
     def _get_anim_delay(self):
         return self._anim_delay
 
     def _set_anim_delay(self, x):
-        if self._anim_delay == x:
+        if self._anim_delay == x or self._durations:
             return
         self._anim_delay = x
         if self._anim_available:
             if self._anim_ev is not None:
                 self._anim_ev.cancel()
                 self._anim_ev = None
 
-            if self._anim_delay >= 0:
+            # start animation only if durations is empty,
+            # otherwise start animation in _set_durations()
+            if self._anim_delay >= 0 and not self._durations:
                 self._anim_ev = Clock.schedule_interval(self._anim,
                                                         self._anim_delay)
 
     anim_delay = property(_get_anim_delay, _set_anim_delay)
+
+    def _get_durations(self):
+        return self._durations
+
+    def _set_durations(self, x):
```

Comment:
i'd rename `x` to `durations` or maybe "value" if you don't mind (i think i prefere `durations`), short variables names are only ok in very small contexts or when their name is really explicit.

Reply:
Yes, of course.